### PR TITLE
Move notary to its new location

### LINKED ISSUE #643: Notary repo moving
With acceptance of Notary in to CNCF, it is moving to a new home. Going forward it will live at `GitHub.com/theupdateframework/notary` and all imports and vendoring should be updated to reflect that new location.

The repo will be moved using GH's tools so a redirect will be in place allowing the old location to continue working, but people vendoring for example, docker cli _and_ notary directly may run in to issues, so the imports should be updated ASAP.

I will be moving notary in the next 48 hours and I'm happy to take on the task of updating this repo to reflect the new location.

### PR DESCRIPTION
**Pull request by thaJeztah**
The https://github.com/docker/notary repository has moved to https://github.com/theupdateframework/notary

fixes #643

full diff; https://github.com/theupdateframework/notary/compare/5d55a30c1bec010a8c6df4c09889acfb4e0a7942...05985dc5d1c71ee6c387e9cd276a00b9d424af53

### CODE REVIEW

File: vendor.conf [hunk 1]
```diff
@@ -40,6 +39,7 @@ github.com/sirupsen/logrus v1.0.3
 github.com/spf13/cobra 7b2c5ac9fc04fc5efafb60700713d4fa609b777b
 github.com/spf13/pflag 97afa5e7ca8a08a383cb259e06636b5e2cc7897f
 github.com/stretchr/testify 4d4bfba8f1d1027c4fdbe371823030df51419987
+github.com/theupdateframework/notary 8a1de3cfc3f1408e54d6364fc949214a4883a9f3
```
Comment (riyazdf) [inline vendor.conf]: this needs to be updated to `05985dc5d1c71ee6c387e9cd276a00b9d424af53` (the commit that moves the import paths) or later cc @endophage

Comment (thaJeztah) [reply vendor.conf]: Thanks! Forgot about references inside notary itself 🤦‍♂️

**Comment by endophage** [comment]
@thaJeztah thanks for doing this before I got to it!

**Comment by thaJeztah** [comment]
You're welcome! 👍

**Pull request merged**